src/blocktypes/devshells.nix

```diff
@@ -36,7 +41,13 @@
             "--builders-use-substitutes"
           )
           nix build "''${nix_args[@]}" --profile "$profile_path/shell-profile"
-          bash -c "source $profile_path/shell-profile/env.bash; SHLVL=$SHLVL; __devshell-motd; exec $SHELL -i"
+          eval "$(nix print-dev-env ${developDrv})"
+          if declare -F __devshell-motd &>/dev/null; then
+            __devshell-motd
+          fi
+          if ! [[ -v STD_DIRENV ]]; then
+            exec $SHELL -i
+          fi
```

Comment:
Did you validate that you end up with the right `SHLVL` in direnv and std enter scenarios?

I can't infer how this may be obsoleted by the adjacent code changes.